addons/binding/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -89,38 +90,34 @@
         public void run() {
             try {
                 pollingLock.lock();
-                try {
-                    if (!lastBridgeConnectionState) {
-                        // if user is not set in configuration try to create a new user on Hue bridge
-                        if (hueBridgeConfig.getUserName() == null) {
-                            hueBridge.getFullConfig();
-                        }
-                        lastBridgeConnectionState = tryResumeBridgeConnection();
-                    }
-                    if (lastBridgeConnectionState) {
-                        doConnectedRun();
+                if (!lastBridgeConnectionState) {
+                    // if user is not set in configuration try to create a new user on Hue bridge
+                    if (hueBridgeConfig.getUserName() == null) {
+                        hueBridge.getFullConfig();
                     }
-                } catch (UnauthorizedException | IllegalStateException e) {
-                    if (isReachable(hueBridge.getIPAddress())) {
+                    lastBridgeConnectionState = tryResumeBridgeConnection();
+                }
+                if (lastBridgeConnectionState) {
+                    doConnectedRun();
+                }
+            } catch (UnauthorizedException | IllegalStateException e) {
+                if (isReachable(hueBridge.getIPAddress())) {
+                    lastBridgeConnectionState = false;
+                    onNotAuthenticated();
+                } else {
+                    if (lastBridgeConnectionState || thing.getStatus() == ThingStatus.INITIALIZING) {
                         lastBridgeConnectionState = false;
-                        onNotAuthenticated();
-                    } else {
-                        if (lastBridgeConnectionState || thing.getStatus() == ThingStatus.INITIALIZING) {
-                            lastBridgeConnectionState = false;
-                            onConnectionLost();
-                        }
+                        onConnectionLost();
                     }
-                } catch (Exception e) {
-                    if (hueBridge != null) {
-                        if (lastBridgeConnectionState) {
-                            logger.debug("Connection to Hue Bridge {} lost.", hueBridge.getIPAddress());
-                            lastBridgeConnectionState = false;
-                            onConnectionLost();
-                        }
+                }
+            } catch (ApiException | IOException e) {
+                if (hueBridge != null) {
```

Comment:
While your at it, maybe combine these 2 if statements to 1.
```suggestion
                if (hueBridge != null && lastBridgeConnectionState) {
```